via/services/youtube_transcript.py

```diff
@@ -0,0 +1,135 @@
+import re
+from base64 import b64encode
+from dataclasses import dataclass
+from typing import Dict, List
+from xml.etree import ElementTree
+
+from via.services.http import HTTPService
+
+
+@dataclass
+class TranscriptInfo:
+    language_code: str
+    name: str
+    url: str
+    autogenerated: bool
+
+    @property
+    def id(self) -> str:  # pylint:disable=invalid-name
+        """Return a unique ID for this transcript."""
+        name = b64encode(self.name.encode("utf-8")).decode("utf-8")
+
+        return ".".join(
+            part or ""
+            for part in [
+                self.language_code,
+                "a" if self.autogenerated else None,
+                name,
+            ]
+        ).rstrip(".")
```

Comment:
If we ever need to we can actually re-compute the values of `TranscriptInfo.language_code`, `TranscriptInfo.name` and `TranscriptInfo.autogenerated` from just the value of `TranscriptInfo.id` (which is the only thing that gets saved to the DB). We wouldn't be able to re-compute the download `url` from the `id`.